src/Metamaps/DataFetcher.js

```diff
@@ -0,0 +1,185 @@
+function get(url) {
+  return fetch(url, {
+    credentials: 'same-origin',
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  })
+}
+
+function post(url, data = {}) {
+  return fetch(url, {
+    credentials: 'same-origin',
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(data)
+  })
+}
+
+function postNoStringify(url, data = {}) {
+  return fetch(url, {
+    credentials: 'same-origin',
+    method: 'POST',
+    body: data
+  })
+}
+
+function put(url, data = {}) {
+  return fetch(url, {
+    credentials: 'same-origin',
+    method: 'PUT',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(data)
+  })
+}
+
+function putNoStringify(url, data = {}) {
+  return fetch(url, {
+    credentials: 'same-origin',
+    method: 'PUT',
+    body: data
+  })
+}
+
+function deleteReq(url) {
+  return fetch(url, {
+    credentials: 'same-origin',
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  })
+}
+
+async function getMetacodes() {
+  const res = await get('/metacodes')
+  const data = await res.json()
+  return data
+}
```

Comment:
I'm fine with this file as-is but I'd prefer to unify the code on Promises or async/await, and I think promises are more flexible.  I'm not sure if it works this way, but it would be cool to use async as an annotation that indicates a function returns a promise. So I'd prefer

```
async function getMetacodes() {
  return get('/metacodes')
    .then(response => response.json())
}
```

I'm not too dogmatic about this though - if you like the look of async/await better, let's try it out